fluent-package/apt/systemd-test/test.sh

```diff
@@ -2,50 +2,73 @@
 
 if [ -z $1 ]; then
     echo "Error: Need to specify lxc image name."
-    echo "Ex.) $ ./test.sh ubuntu:20.04"
+    echo "Ex.) $ ./test.sh ubuntu:20.04 "
+    echo "Ex. CI) $ ./test.sh ubuntu:20.04 install-newly.sh local"
     exit 1
 fi
 
 image=$1
+test_file=$2
+apt_repo_type=$3
 dir="/host/fluent-package/apt/systemd-test"
 
 set -eux
 
-test_filenames=(
-    update-to-next-version.sh
-)
-
-if [ ! $image = "images:debian/12" ]; then
-    # As no bookworm package for v4, so execute upgrade test for other code name.
-    test_filenames+=(
-        update-from-v4.sh
-        update-to-next-version-with-backward-compat-for-v4.sh
-        downgrade-to-v4.sh
-    )
-fi
-
-for apt_repo_type in local v5 lts; do
-    echo -e "\nRun test: $apt_repo_type\n"
+if [ $CI = "true" ]; then
+    echo "::group::Run test: launch $image"
     lxc launch $image target
     sleep 5
+    echo "::endgroup::"
+    echo "::group::Run test: configure $image"
     lxc config device add target host disk source=$PWD path=/host
     lxc list
+    echo "::endgroup::"
+    echo "::group::Run test: setup $image"
     lxc exec target -- $dir/setup.sh
-    lxc exec target -- $dir/install-newly.sh $apt_repo_type
+    echo "::endgroup::"
+    echo "::group::Run test: $test_file $apt_repo_type on $image"
+    lxc exec target -- $dir/$test_file $apt_repo_type
+    echo "::endgroup::"
+    echo "::group::Run test: cleanup $image"
     lxc stop target
     lxc delete target
-done
+    echo "::endgroup::"
+else
+    test_filenames=(
+	update-to-next-version.sh
+    )
 
-for test_filename in ${test_filenames[@]}; do
-    echo -e "\nRun test: $test_filename\n"
-    lxc launch $image target
-    sleep 5
-    lxc config device add target host disk source=$PWD path=/host
-    lxc list
-    lxc exec target -- $dir/setup.sh
-    lxc exec target -- $dir/$test_filename
-    lxc stop target
-    lxc delete target
-done
+    if [ ! $image = "images:debian/12" ]; then
+	# As no bookworm package for v4, so execute upgrade test for other code name.
+	test_filenames+=(
+            update-from-v4.sh
+            update-to-next-version-with-backward-compat-for-v4.sh
+            downgrade-to-v4.sh
+	)
+    fi
+
+    for apt_repo_type in local v5 lts; do
+	echo -e "\nRun test: $apt_repo_type\n"
+	lxc launch $image target
+	sleep 5
+	lxc config device add target host disk source=$PWD path=/host
+	lxc list
+	lxc exec target -- $dir/setup.sh
+	lxc exec target -- $dir/install-newly.sh $apt_repo_type
+	lxc stop target
+	lxc delete target
+    done
 
+    for test_filename in ${test_filenames[@]}; do
+	echo -e "\nRun test: $test_filename\n"
+	lxc launch $image target
+	sleep 5
+	lxc config device add target host disk source=$PWD path=/host
+	lxc list
+	lxc exec target -- $dir/setup.sh
+	lxc exec target -- $dir/$test_filename
+	lxc stop target
+	lxc delete target
+    done
+fi
```

Comment:
I'm sorry if I am confusing you.
If you don't need this logic for the not CI case, then I think we can remove it.
It is because, now, I think as follows,

> Sorry, I don't think we need to worry too much about this point.
When I run tests locally, I'm likely to run specific tests repeatedly.
It may be enough to use CI to run all tests.

and it would be annoying to maintain a list of test files on both the CI side and the script side.

If you need it, no problem.
If it's because of my previous suggestion, we should remove it and simplify the script.

I'm sorry if I am confusing you.